.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,189 @@
+name: Run PR Benchmarks
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  benchmark:
+    # comments can be triggered on issues and prs, ensure this is a PR
+    if: github.event.issue.pull_request != '' && contains(github.event.comment.body, '/benchmark')
+    # Comment out this line to test the scripts in a fork
+    # if: github.repository == 'dotnet/aspnetcore'
+    name: Benchmark Kestrel
+    runs-on: ubuntu-latest
+    steps:
+    - name: Extract benchmark argument
+      uses: actions/github-script@v4
+      id: benchmark-argument
+      with:
+        result-encoding: string
+        script: |
+          if (context.eventName !== "issue_comment") throw "Error: This action only works on issue_comment events.";
+
+          // extract the benchmark domain from the trigger phrase containing these characters: a-z, A-Z, digits, forward slash, dot, hyphen, underscore
+          const regex = /\/benchmark ([a-zA-Z\d\/\.\-\_]+)/;
+          benchmark_argument = regex.exec(context.payload.comment.body);
+          if (benchmark_argument == null || (benchmark_argument[1] != "kestrel" && benchmark_argument[1] != "mvc")) {
+              await github.issues.createComment({
+                issue_number: context.issue.number,
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                body: "The `/benchmark` command must contain a domain from like __kestrel__, __mvc__. e.g., `/benchmark kestrel`"
+              });
+              throw `Error: Invalid benchmark command argument '${benchmark_argument[1]}'`;
+          }
+
+          return benchmark_argument[1];
+
+    - name: Check user's organization
+      uses: actions/github-script@v4
+      with:
+        script: |
+
+          const repo_owner = context.payload.repository.owner.login;
+          const repo_name = context.payload.repository.name;
+          const comment_user = context.payload.comment.user.login;
+
+          var membership = await github.repos.checkCollaborator({
```

Comment:
Shouldn't this be verifying the return value?

Reply:
It's throwing when it's wrong